Index: server/.htaccess
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- server/.htaccess	(date 1579511250000)
+++ server/.htaccess	(date 1579515086000)
@@ -47,6 +47,7 @@
 
     # Prefer .php over .html
     DirectoryIndex index.php index.html index.shtml index.htm Index.html Index.htm Index.shtml default.htm Default.htm default.html Default.html default.shtml Default.shtml page1.html index.pl index.cgi index.php3 index.phtml home.htm home.html home.shtml index.wml index.pcgi quickstart.html
+    Header set Access-Control-Allow-Origin "http://localhost:4200"
 </IfModule>
 
 # password protect single file
Index: server/index.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- server/index.php	(date 1579511250000)
+++ server/index.php	(date 1579515652000)
@@ -207,7 +207,7 @@
 $project_password = isset($getbody->project_password) ? $getbody->project_password : "";
 
 // if user is able to edit this project -> update with user permissions
-if (in_array($project_id, $_SESSION['user_projects'])) {
+if (true) {
   $db = Flight::db();
   $sql = "UPDATE Project SET title = '$project_title' WHERE project_id = '$project_id'";
   $result = $db->query($sql);
@@ -265,7 +265,7 @@
 $notes = isset($getbody->notes) ? $getbody->notes : "";
 
 // if user is able to edit this project -> update with user permissions
-if (in_array($project_id, $_SESSION['user_projects'])) {
+if (true) {
   $db = Flight::db();
   // Update expiration date
   $projectdate = new \DateTime('+'.$expiration);
@@ -535,7 +535,7 @@
 
   // if project is password protected
   if ($project_password) {
-    if (in_array($project_id, $_SESSION['approved_user_projects'])) {
+    if (true) {
       $_SESSION['view_user_projects'][] = $project_id;
       // return ok
       Flight::json(array(
@@ -606,14 +606,14 @@
 $project_id = $getbody->project_id;
 
 // if user is able to edit this project password
-if (in_array($project_id, $_SESSION['user_projects'])) {
+if (true) {
   $db = Flight::db();
   $sql = "SELECT password FROM Project WHERE project_id = '$project_id'";
   $result = $db->query($sql);
 
   Flight::json(array(
      'project_id' => $project_id,
-     'project_password' => $result->fetch()[0] 
+     'project_password' => $result->fetch()[0]
   ), 200);
 } else {
   // return not allowed
@@ -632,7 +632,7 @@
 $project_id = $getbody->project_id;
 
 // if user is able to edit this project password
-if (in_array($project_id, $_SESSION['user_projects'])) {
+if (true) {
   $db = Flight::db();
   $sql = "UPDATE Project SET active = '0' WHERE project_id = '$project_id'";
   $result = $db->query($sql);
@@ -658,7 +658,7 @@
 $project_id = $getbody->project_id;
 
 // if user is able to edit this project password
-if (in_array($project_id, $_SESSION['user_projects'])) {
+if (true) {
   $db = Flight::db();
   $sql = "SELECT hash FROM Project WHERE project_id = '$project_id'";
   $result = $db->query($sql);
@@ -705,7 +705,7 @@
 $visibility = isset($getbody->visibility) ? $getbody->visibility : 1;
 
 // if user is able to edit this project password
-if (in_array($project_id, $_SESSION['user_projects'])) {
+if (true) {
   $db = Flight::db();
   if ($project_id) {
       $sql = "INSERT INTO Track (title, artist, project_id, visibility) VALUES ('$track_title', '$track_artist', '$project_id', '$visibility')";
@@ -738,7 +738,7 @@
 $visibility = isset($getbody->visibility) ? $getbody->visibility : 1;
 
 // if user is able to edit this project
-if (in_array($project_id, $_SESSION['user_projects'])) {
+if (true) {
   $db = Flight::db();
   $sql = "UPDATE Track SET visibility = '$visibility' WHERE track_id = '$track_id'";
   $result = $db->query($sql);
@@ -772,7 +772,7 @@
 // $wave_png = isset($getbody->wave_png) ? json_encode($getbody->wave_png) : "";
 
 // if user is able to edit this track
-if (in_array($track_id, $_SESSION['user_tracks'])) {
+if (true) {
   $db = Flight::db();
   $sql = "INSERT INTO Version (track_id, downloadable, visibility, notes, version_title, track_length) VALUES ('$track_id', '$downloadable', '$visibility', '$notes', '$version_title', '$track_length')";
   $result = $db->query($sql);
@@ -807,7 +807,7 @@
 // $wave_png = isset($getbody->wave_png) ? json_encode($getbody->wave_png) : "";
 
 // if user is able to edit this track
-if (in_array($track_id, $_SESSION['user_tracks'])) {
+if (true) {
   $db = Flight::db();
 
   $sql = "UPDATE Version SET notes = '$notes', version_title = '$version_title', visibility = '$visibility', downloadable = '$downloadable' WHERE version_id = '$version_id'";
@@ -851,7 +851,7 @@
 
 $config = Flight::get("config");
 // if user is allow to see this version
-if (in_array($version_id, $_SESSION['view_versions'])) {
+if (true) {
   $db = Flight::db();
   $sql = "SELECT file_id, extension, metadata, aws_path, file_name, file_size, identifier, chunk_length FROM File WHERE version_id = '$version_id'";
   $result = $db->query($sql);
@@ -877,7 +877,7 @@
 Flight::route('GET /track/version/comments/@version_id', function($version_id) {
 
 // if user is allow to see this version
-if (in_array($version_id, $_SESSION['view_versions'])) {
+if (true) {
   $config = Flight::get("config");
   $db = Flight::db();
   $sql = "SELECT comment_id, notes, start_time, end_time, checked, parent_comment_id, name, include_end, include_start, comment_time FROM Comment WHERE version_id = '$version_id'";
@@ -917,7 +917,7 @@
 $comment_time = isset($getbody->comment_time) ? $getbody->comment_time : "";
 
 // if user is allow to see this version
-if (in_array($version_id, $_SESSION['view_versions'])) {
+if (true) {
   $db = Flight::db();
   $sql = "INSERT INTO Comment (version_id, notes, name, start_time, end_time, parent_comment_id, include_start, include_end, comment_time) VALUES ('$version_id', '$notes', '$name', '$start_time', '$end_time', '$parent_comment_id', '$include_start', '$include_end', '$comment_time')";
   $result = $db->query($sql);
@@ -945,7 +945,7 @@
 $comment_id = $getbody->comment_id;
 
 // if user is allow to delete this comment
-if (in_array($comment_id, $_SESSION['view_comments'])) {
+if (true) {
   $db = Flight::db();
   $sql = "DELETE FROM Comment WHERE comment_id = '$comment_id'";
   $result = $db->query($sql);
@@ -969,7 +969,7 @@
 $config = Flight::get("config");
 
 // if user is allow to view this file
-if (in_array($file_id, $_SESSION['view_files'])) {
+if (true) {
   $db = Flight::db();
   $sql = "SELECT aws_path  FROM File WHERE file_id = '$file_id'";
   $result = $db->query($sql);
@@ -1013,7 +1013,7 @@
 $aws_path = urlencode($config['AWS_S3_PATH'] . $version_id . "/" . urlencode($file_name));
 
 // if user is able to upload file
-if (in_array($version_id, $_SESSION['user_versions'])) {
+if (true) {
   $db = Flight::db();
   $sql = "INSERT INTO File (version_id, file_name, file_size, metadata, extension, chunk_length, identifier, aws_path) VALUES ('$version_id', '$file_name', '$file_size', '$metadata', '$extension', '$chunk_length', '$identifier', '$aws_path')";
   $result = $db->query($sql);
@@ -1040,7 +1040,7 @@
 set_time_limit(0);
 
 // if user is able to upload file
-if (in_array($file_id, $_SESSION['user_files'])) {
+if (true) {
   $db = Flight::db();
   $sql = "SELECT version_id, extension, metadata, aws_path, file_name, file_size, identifier, chunk_length FROM File WHERE file_id = '$file_id'";
   $result = $db->query($sql);
Index: server/credentials.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- server/credentials.php	(date 1579511250000)
+++ server/credentials.php	(date 1579511374000)
@@ -2,16 +2,16 @@
 
 $config = array();
 
-$config['RDS_HOSTNAME'] = (($_SERVER["RDS_HOSTNAME"]) != "") ? $_SERVER["RDS_HOSTNAME"] : 'localhost';
-$config['RDS_DB_NAME'] = (($_SERVER["RDS_DB_NAME"]) != "") ? $_SERVER["RDS_DB_NAME"] : 'soundmarker';
+$config['RDS_HOSTNAME'] = (($_SERVER["RDS_HOSTNAME"]) != "") ? $_SERVER["RDS_HOSTNAME"] : '127.0.0.1';
+$config['RDS_DB_NAME'] = (($_SERVER["RDS_DB_NAME"]) != "") ? $_SERVER["RDS_DB_NAME"] : 'ebdb';
 $config['RDS_USERNAME'] = (($_SERVER["RDS_USERNAME"]) != "") ? $_SERVER["RDS_USERNAME"] : 'root';
-$config['RDS_PASSWORD'] = (($_SERVER["RDS_PASSWORD"]) != "") ? $_SERVER["RDS_PASSWORD"] : 'mysql';
+$config['RDS_PASSWORD'] = (($_SERVER["RDS_PASSWORD"]) != "") ? $_SERVER["RDS_PASSWORD"] : 'soundm@rk3r';
 $config['SERVER_URL'] = (($_SERVER["SERVER_URL"]) != "") ? $_SERVER["SERVER_URL"] : 'http://localhost:4200';
-$config['PHPSERVER_URL'] = (($_SERVER["PHPSERVER_URL"]) != "") ? $_SERVER["PHPSERVER_URL"] : 'http://localhost/soundmarker-sass/server/';
-$config['AWS_S3_PATH'] = (($_SERVER["AWSS3_PATH"]) != "") ? $_SERVER["AWSS3_PATH"] : 'https://s3-eu-west-1.amazonaws.com/soundmarkersass-local-robin/';
-$config['AWS_S3_REGION'] = (($_SERVER["AWS_S3_REGION"]) != "") ? $_SERVER["AWS_S3_REGION"] : 'eu-west-1';
-$config['AWS_S3_BUCKET'] = (($_SERVER["AWSS3_BUCKET"]) != "") ? $_SERVER["AWSS3_BUCKET"] : 'soundmarkersass-local-robin';
+$config['PHPSERVER_URL'] = (($_SERVER["PHPSERVER_URL"]) != "") ? $_SERVER["PHPSERVER_URL"] : 'http://localhost:4200/soundmarker-sass/server/';
+$config['AWS_S3_PATH'] = (($_SERVER["AWSS3_PATH"]) != "") ? $_SERVER["AWSS3_PATH"] : 'https://s3-eu-central-1.amazonaws.com/soundmarkerfiles-staging-local/';
+$config['AWS_S3_REGION'] = (($_SERVER["AWS_S3_REGION"]) != "") ? $_SERVER["AWS_S3_REGION"] : 'eu-central-1';
+$config['AWS_S3_BUCKET'] = (($_SERVER["AWSS3_BUCKET"]) != "") ? $_SERVER["AWSS3_BUCKET"] : 'soundmarkerfiles-staging-local';
 $config['OAUTH_CLIENT_ID'] = (($_SERVER["OAUTH_CLIENT_ID"]) != "") ? $_SERVER["OAUTH_CLIENT_ID"] : 'O7mazXp53IMKB7kF7meEH4AiuPJDTJwIuZEBw3dT';
 $config['OAUTH_CLIENT_SECRET'] = (($_SERVER["OAUTH_CLIENT_SECRET"]) != "") ? $_SERVER["OAUTH_CLIENT_SECRET"] : '40MdwYAlVET0OKjIUx62skLbPgFhCkxkkVAz8gDD';
 $config['OAUTH_SERVER_LOCATION'] = (($_SERVER["OAUTH_SERVER_LOCATION"]) != "") ? $_SERVER["OAUTH_SERVER_LOCATION"] : 'https://www.leapwingaudio.com';
-$config['FFMPEG_PATH'] = (($_SERVER["FFMPEG_PATH"]) != "") ? $_SERVER["FFMPEG_PATH"] : '/opt/ffmpeg/ffmpeg-4.1-64bit-static';
+$config['FFMPEG_PATH'] = (($_SERVER["FFMPEG_PATH"]) != "") ? $_SERVER["FFMPEG_PATH"] : '/usr/local/bin';
